src/test/java/jenkins/advancedqueue/jobinclusion/strategy/JobInclusionJobPropertyTest.java

```diff
@@ -3,53 +3,238 @@
+        assertTrue(jobProperty.isUseJobGroup());
+        assertNull(jobProperty.getJobGroupName());
+
+        // Create a JobInclusionJobProperty with useJobGroup set to false and jobGroupName set to null
+        JobInclusionJobProperty jobPropertyFalse = new JobInclusionJobProperty(false, null);
+        assertFalse(jobPropertyFalse.isUseJobGroup());
+        assertNull(jobPropertyFalse.getJobGroupName());
+
+        // Create a JobInclusionJobProperty with useJobGroup set to true and jobGroupName set to a non-null value
+        JobInclusionJobProperty jobPropertyWithGroupName = new JobInclusionJobProperty(true, "testJobGroupName");
+        assertTrue(jobPropertyWithGroupName.isUseJobGroup());
+        assertEquals("testJobGroupName", jobPropertyWithGroupName.getJobGroupName());
     }
 
     @Test
-    public void getDisplayNameTest() {
+    public void descriptorImplGetDisplayName() {
+        JobInclusionJobProperty.DescriptorImpl descriptor = new JobInclusionJobProperty.DescriptorImpl();
         assertEquals("XXX", descriptor.getDisplayName());
     }
 
     @Test
-    public void getJobGroupsTest() {
-        assertNotNull(descriptor.getJobGroups());
+    public void getDescriptorTest() {
+        JobInclusionJobProperty jobProperty = new JobInclusionJobProperty(true, null);
+        JobInclusionJobProperty.DescriptorImpl descriptor = jobProperty.getDescriptor();
+
+        // Verify that the descriptor is not null
+        assertNotNull(descriptor);
+
+        // Verify the display name of the descriptor
+        assertEquals("XXX", descriptor.getDisplayName());
+
+        // Verify the type of the descriptor
+        assertTrue(descriptor instanceof JobInclusionJobProperty.DescriptorImpl);
+
+        // Verify the isUsed method of the descriptor
+        assertFalse(descriptor.isUsed());
     }
 
     @Test
-    public void isUsedTest() {
+    public void descriptorImplIsUsed() {
+        JobInclusionJobProperty.DescriptorImpl descriptor = new JobInclusionJobProperty.DescriptorImpl();
         assertFalse(descriptor.isUsed());
     }
+
+    @Test
+    public void getJobGroupsTest() {
+        JobInclusionJobProperty.DescriptorImpl descriptor = new JobInclusionJobProperty.DescriptorImpl();
+        ListBoxModel jobGroups = descriptor.getJobGroups();
+
+        jobGroups.add("testJobGroup", "testGroup");
+        // Verify that the jobGroups is not null
+        assertNotNull(jobGroups);
+
+        // Verify that the jobGroups is not empty
+        assertFalse(jobGroups.isEmpty());
+
+        // Verify the size of the jobGroups
+        assertEquals("Expected number of job groups", 1, jobGroups.size());
+
+        // Verify the properties of the first job group
+        ListBoxModel.Option firstOption = jobGroups.get(0);
+        assertNotNull(firstOption);
+        assertEquals("Expected display name", "testJobGroup", firstOption.name);
+        assertEquals("Expected value", "testGroup", firstOption.value);
+    }
```

Comment:
This test seems to be only testing the additions to the JobInclusionJobProperty object.  As far as I can tell, those additions are not needed.
```suggestion
```